### PR TITLE
Use music objects instead of IDs

### PR DESCRIPTION
**Pull request by Gymnasiast**
Backported from the NSF.

(This one could cut quite a lot of files from the NSF diff!)

Do not squash merge!

### CODE REVIEW

File: src/openrct2/rct1/S4Importer.cpp [hunk 1]
```diff
@@ -1014,8 +1022,20 @@ namespace RCT1
                     dst->track_colour[i].additional = RCT1::GetColour(src->track_colour_additional[i]);
                     dst->track_colour[i].supports = RCT1::GetColour(src->track_colour_supports[i]);
                 }
+            }
+
+            dst->entrance_style = OBJECT_ENTRY_INDEX_NULL;
```
Comment (Gymnasiast) [inline src/openrct2/rct1/S4Importer.cpp:1027]: revert this bit

File: src/openrct2/ride/gentle/meta/Circus.h [hunk 2]
```diff
@@ -43,7 +43,7 @@ constexpr const RideTypeDescriptor CircusRTD =
     SET_FIELD(UpkeepCosts, { 50, 1, 0, 0, 0, 0 }),
     SET_FIELD(BuildCosts, { 125, 2, 1, }),
     SET_FIELD(DefaultPrices, { 15, 0 }),
-    SET_FIELD(DefaultMusic, MUSIC_STYLE_CIRCUS_SHOW),
+    SET_FIELD(DefaultMusic, {}),
```
Comment (Gymnasiast) [inline src/openrct2/ride/gentle/meta/Circus.h:46]: As I expected, this breaks the circus sounds.

**Review by Broxzier — COMMENTED** [review]
Haven't checked every single line, but the changes look good to me. One thing I did find while testing is that Custom 1 and Custom 2 did show up in the dropdown menu after adding the, but nothing was played when selecting them. I had to restart the game before they worked. This was probably already an issue before, not introduced by this PR.

**Review by Broxzier — APPROVED** [review]
Just confirmed that the above issue was already present before this PR.

File: src/openrct2/rct2/S6Importer.cpp [hunk 3]
```diff
@@ -800,7 +800,12 @@ class S6Importer final : public IParkImporter
             }
         }
 
-        dst->music = src->music;
+        auto musicStyle = OBJECT_ENTRY_INDEX_NULL;
+        if (GetRideTypeDescriptor(dst->type).HasFlag(RIDE_TYPE_FLAG_ALLOW_MUSIC))
+        {
+            musicStyle = src->music;
+        }
+        dst->music = musicStyle;
```
Comment (duncanspumpkin) [inline src/openrct2/rct2/S6Importer.cpp:808]: I dont see the reverse in S6Exporter. I assume its not required as it truncates fine?

**Pull request merged**